jira_playbooks/tasks/create_issue.yml

```diff
@@ -0,0 +1,91 @@
+---
+# github issue can be in the following formats
+# * repo/number e.g. network/316
+# * org/repo/number e.g. linux-system-roles/network/316
+# * url e.g. https://github.com/linux-system-roles/network/pull/316
+- name: Reset issue content
+  set_fact:
+    __gh_issue_content: {}
+    __gh_roles: ''
+
+- name: Get github issue/pr
+  when:
+    - github_issue | d("") | length > 0
+    - create_from_github_issue | d(true)
+  vars:
+    __gh_issue_ary: "{{ github_issue | split('/') }}"
+    __gh_api_base: https://api.github.com/repos/
+    __gh_org: "{{ __gh_issue_ary[3] if __gh_issue_ary | length > 4
+      else __gh_issue_ary[0] if __gh_issue_ary | length > 2
+      else 'linux-system-roles' }}"
+    __gh_repo: "{{ __gh_issue_ary[4] if __gh_issue_ary | length > 4
+      else __gh_issue_ary[1] if __gh_issue_ary | length > 2
+      else __gh_issue_ary[0] }}"
+    __gh_num: "{{ __gh_issue_ary[-1] }}"
+    __gh_api_url: "{{ __gh_api_base ~ __gh_org ~ '/' ~ __gh_repo ~ '/issues/' ~
+      __gh_num }}"
+  block:
+    - name: Get issue/pr from url
+      uri:
+        url: "{{ __gh_api_url }}"
+        method: GET
+      register: __gh_issue
+
+    - name: Set facts 1
+      set_fact:
+        __gh_roles: "{{ __gh_repo }}"
+        __gh_issue_type: "{{ 'pull' if 'pull_request' in __gh_issue.json
+          else 'issue' }}"
+
+    - name: Set facts 2
+      set_fact:
+        __gh_issue_url: "{{ github_issue
```

Comment:
Unfortunately no, because `__gh_issue_url` uses `__gh_issue_type` - the way `set_fact` works is that it evaluates the variables immediately, so you'll get an undefined variable error for `__gh_issue_type`.